src/pyhf/infer/mle.py

```diff
@@ -47,6 +47,14 @@ def twice_nll(pars, data, pdf):
     return -2 * pdf.logpdf(pars, data)
 
 
+def _validate_fit_inputs(init, bounds, fixed_params):
+    for i, (value, bound) in enumerate(zip(init, bounds)):
+        if not (bound[0] <= value <= bound[1]):
+            raise ValueError(
+                f'parameter {i}  value {value} of hypotest lies outside of bounds: {bound}'
```

Comment:
It would be good if we could make this error message give a bit more information on the offending parameter. It additionally isn't restricted to `hypotest` only, but to any fit.